JesusFilm/core
# fix: add loading state to card templates

# Description

### Issue

No loading state for adding card templates

[Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/37071143/todos/7439533560)

### Solution

Add skeleton on loading state

# External Changes

* Refactor test spec files for each card template
* Add loading state to storybook
* Add more unit tests

# Additional information

n/a

# How to test
Consider turning browser throttling on for testing.
1. Create a new step and open it in the editor mode. When presented with the card template drawer, try and mash one of the templates multiple times. You should not be able to because the templates should disable.

2. Create a new step and open it in the editor mode. When presented with the card template drawer, try and quickly select multiple card templates. You should not be able to because the templates should disable.

# Acceptance Criteria
As a user I should not be able to add a card template multiple times
As a user I should not be able to add multiple card templates to the same step
